Assets/Scripts/SS3D/Systems/Inventory/Containers/Hand.cs

```diff
@@ -0,0 +1,145 @@
+﻿using SS3D.Systems.Inventory.Containers;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using SS3D.Systems.Inventory.Items;
+using System.Linq;
+using SS3D.Interactions.Interfaces;
+using SS3D.Interactions;
+using FishNet.Object;
+using static SS3D.Systems.Inventory.Containers.AttachedContainer;
+
+namespace SS3D.Systems.Inventory.Containers
+{
+	/// <summary>
+	/// A hand is what an entity uses to grab and hold items, to interact with things in range.
+	/// </summary>
+	public class Hand : InteractionSource, IInteractionRangeLimit, IInteractionOriginProvider
+	{
+		/// <summary>
+		/// Container linked to this hand, necessary to hold stuff.
+		/// </summary>
+		public AttachedContainer Container;
+
+		/// <summary>
+		/// Horizontal and vertical max distance to interact with stuff.
+		/// </summary>
+		public RangeLimit range = new(1.5f, 2);
+
+		// pickup icon that this hand uses when there's a pickup interaction
+		// TODO: When AssetData is on, we should update this to not use this
+		public Sprite pickupIcon;
+
+		/// <summary>
+		/// The item held in this hand, if it exists
+		/// </summary>
+		public Item ItemInHand => Container.Items.FirstOrDefault();
+
+		/// <summary>
+		/// Point from where distances for interaction is computed.
+		/// </summary>
+		public Transform interactionOrigin;
+
+		/// <summary>
+		/// the hands script controlling this hand.
+		/// </summary>
+		public Hands handsController;
+
+		public Vector3 InteractionOrigin => interactionOrigin.position;
+
+		public delegate void HandHandler(Hand hand);
+		public event HandHandler OnHandDisabled;
+
+		protected override void OnDisabled()
+		{
+			if (!IsServer)
+			{
+				return;
+			}
+
+			OnHandDisabled?.Invoke(this);
+		}
+
+		public bool IsEmpty()
+		{
+			return Container.Container.Empty;
+		}
+
+		/// <summary>
+		/// Get the interaction source from stuff in hand if there's any.
+		/// </summary>
+		/// <returns></returns>
+		public IInteractionSource GetActiveTool()
```

Comment:
Can you explain this? What is interaction source?

Reply:
updated IInteractionSource doc, lmk if this helps enough

Reply:
If a screwdriver can be a source for opening panels, then why interactionSource.Source = this; sets a hand as the source?

Reply:
@iamteapot422 the method check if an item held in hand should be used as the interaction source, such as a screwdriver or a hammer or whatever... If nothing is found, the hand itself is the interaction source, hands have their own set of available interactions (hit, pickup... more coming)

Reply:
@stilnat shouldn't the check be "if (interactionSource == null)" then?